src/libs/web3-data-provider/Web3Provider.tsx

```diff
@@ -149,9 +149,13 @@ export const Web3ContextProvider: React.FC<{ children: ReactElement }> = ({ chil
       // @ts-expect-error isSafeApp not in abstract connector type
       gnosisConnector.isSafeApp().then((loadedInSafe) => {
         if (loadedInSafe) {
-          connectWallet(WalletType.GNOSIS).catch(() => {
-            setTriedSafe(true);
-          });
+          connectWallet(WalletType.GNOSIS)
+            .then(() => {
+              setTriedSafe(true);
+            })
+            .catch(() => {
+              setTriedSafe(true);
```

Comment:
this also true?

Reply:
yes, its to store that it has tried to connect to gnosis. even if it fails (which will happen when you are not accessing the app through gnosis safe)